### PR TITLE
Improve search in organizations & advanced filters with common fuzzy logic

### PR DESCRIPTION
**Pull request by jahow**
This PR introduces a `createFuzzyFilter` util that is used both in organizations & advanced filters search. This filter is only client side and will split compared strings by whitespace and compare each part while ignoring special chars etc.

In the future we might want to use a more advanced fuzzy filtering library.

### CODE REVIEW

File: libs/ui/inputs/src/lib/dropdown-multiselect/dropdown-multiselect.component.ts [hunk 1]
```diff
@@ -74,9 +77,10 @@ export class DropdownMultiselectComponent {
   }
 
   get filteredChoicesByText() {
-    return this.choices.filter((choice) =>
-      choice.label.toLowerCase().includes(this.searchInputValue?.toLowerCase())
-    )
+    const filterPattern = this.searchInputValue?.toLowerCase()
```
Comment (tkohr) [inline libs/ui/inputs/src/lib/dropdown-multiselect/dropdown-multiselect.component.ts:80]: `?.toLowerCase()` might not be necessary anymore here, as the `FuzzyFilter` seems to handle this.

File: libs/util/shared/src/lib/utils/fuzzy-filter.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+type FuzzyFilter = (input: string) => boolean
+
+function normalize(input: string) {
+  return input
+    .toLowerCase()
+    .normalize('NFD') // explode composite chars (e.g. é) into multiple chars
+    .replace(/[\u0300-\u036f]/g, '') // remove accents
+    .replace(/œ/g, 'oe') // remove accents
```
Comment (tkohr) [inline libs/util/shared/src/lib/utils/fuzzy-filter.ts:8]: Looking at french specificities, `æ` is not relevant in modern french anymore, is it?

**Review by tkohr — APPROVED** [review]
Nice @jahow, very elegant solution to make this feature generic and reusable, thanks!

File: libs/util/shared/src/lib/utils/fuzzy-filter.ts (hunk 2 above)
Comment (jahow) [reply libs/util/shared/src/lib/utils/fuzzy-filter.ts:8]: not ae but oe yes, like oeuf :)

File: libs/ui/inputs/src/lib/dropdown-multiselect/dropdown-multiselect.component.ts (hunk 1 above)
Comment (jahow) [reply libs/ui/inputs/src/lib/dropdown-multiselect/dropdown-multiselect.component.ts:80]: right!

File: libs/util/shared/src/lib/utils/fuzzy-filter.ts (hunk 2 above)
Comment (tkohr) [reply libs/util/shared/src/lib/utils/fuzzy-filter.ts:8]: yep, I just saw that æ also ([used to](https://fr.wikipedia.org/wiki/%C3%86)) exist, but seems to have been replaced mostly by "accent aigu" nowadays, if I understand right :-)

Comment (jahow) [reply libs/util/shared/src/lib/utils/fuzzy-filter.ts:8]: To be honest I would have never thought of it, but I encountered a metadata record with it!

**Pull request merged**